### ISSUE TITLE
Reading: Adaptive Compression of Word Embeddings

### ISSUE THREAD

**Issue by a1da4** [a1da4/paper-survey#153]
## 0. Paper
@inproceedings{kim-etal-2020-adaptive,
    title = "Adaptive Compression of Word Embeddings",
    author = "Kim, Yeachan  and
      Kim, Kang-Min  and
      Lee, SangKeun",
    booktitle = "Proceedings of the 58th Annual Meeting of the Association for Computational Linguistics",
    month = jul,
    year = "2020",
    address = "Online",
    publisher = "Association for Computational Linguistics",
    url = "https://www.aclweb.org/anthology/2020.acl-main.364",
    doi = "10.18653/v1/2020.acl-main.364",
    pages = "3950--3959",
    abstract = "Distributed representations of words have been an indispensable component for natural language processing (NLP) tasks. However, the large memory footprint of word embeddings makes it challenging to deploy NLP models to memory-constrained devices (e.g., self-driving cars, mobile devices). In this paper, we propose a novel method to adaptively compress word embeddings. We fundamentally follow a code-book approach that represents words as discrete codes such as (8, 5, 2, 4). However, unlike prior works that assign the same length of codes to all words, we adaptively assign different lengths of codes to each word by learning downstream tasks. The proposed method works in two steps. First, each word directly learns to select its code length in an end-to-end manner by applying the Gumbel-softmax tricks. After selecting the code length, each word learns discrete codes through a neural network with a binary constraint. To showcase the general applicability of the proposed method, we evaluate the performance on four different downstream tasks. Comprehensive evaluation results clearly show that our method is effective and makes the highly compressed word embeddings without hurting the task accuracy. Moreover, we show that our model assigns word to each code-book by considering the significance of tasks.",
}
## 1. What is it?
They propose a compress method to adapt word representations into some devices.

## 3. Where is the key to technologies and techniques?
![スクリーンショット 2021-03-10 1 00 19](https://user-images.githubusercontent.com/45454055/110499859-fe092d80-813b-11eb-8c22-e6ed067abb90.png)
Code-books: discrete vector of word. e.g. dog: (3, 5, 2, 1)
First, they encode a target word vector ew as follows:
![Uploading スクリーンショット 2021-03-10 1.05.43.png…]()

However, when a word vector in the code-book is selected,  this function is not differentiable.
To avoid this problem, they use Gumbel softmax to have a k-dimensional one-hot vector.
![スクリーンショット 2021-03-10 1 08 40](https://user-images.githubusercontent.com/45454055/110501070-28a7b600-813d-11eb-80b0-d3e87d8a056e.png)

Second, binary code learning is conducted as follows:
![スクリーンショット 2021-03-10 1 18 39](https://user-images.githubusercontent.com/45454055/110502455-8e487200-813e-11eb-83a6-f2d86a2d6d69.png)

Finally, a target vector ow is calculated from these vectors.
![スクリーンショット 2021-03-10 1 19 27](https://user-images.githubusercontent.com/45454055/110502577-ab7d4080-813e-11eb-8ebc-662d41e8a09c.png)

## 4. How did evaluate it?
They compared their method and previous methods by four tasks.
* sentiment classification: SST-5
* chunking: CoNLL-2000
* textual entailment: SNLI
* language modeling: PTB

![スクリーンショット 2021-03-10 1 22 41](https://user-images.githubusercontent.com/45454055/110503034-1fb7e400-813f-11eb-9d0a-6e044b2c42b8.png)

## 5. Is there a discussion?
They visualize compressed word embeddings in the sentiment classification task.
This figure shows that important words (positive, negative) are longer code-book.
![スクリーンショット 2021-03-10 1 26 03](https://user-images.githubusercontent.com/45454055/110503592-b1275600-813f-11eb-9635-bb91bb6b8e0b.png)
